bip-schnorr/reference.py

```diff
@@ -88,6 +89,49 @@ def schnorr_verify(msg, pubkey, sig):
         return False
     return True
 
+def schnorr_batch_verify(msgs, pubkeys, sigs):
+    if len(msgs) != len(pubkeys) or len(pubkeys) != len(sigs):
+        raise ValueError('All the parameters must be the same length.')
+    if len(msgs) < 1:
+        raise ValueError('All the parameters must have at least one element.')
+
+    i, ls, rs = 0, 0, None
+    for sig in sigs:
+        msg = msgs[i]
+        pubkey = pubkeys[i]
+        if len(msg) != 32:
+            raise ValueError('The message must be a 32-byte array.')
+        if len(pubkey) != 33:
+            raise ValueError('The public key must be a 33-byte array.')
+        if len(sig) != 64:
+            raise ValueError('The signature must be a 64-byte array.')
+        P = point_from_bytes(pubkey)
+        if (P is None):
+            return False
+        r = int_from_bytes(sig[0:32])
+        s = int_from_bytes(sig[32:64])
+        if (r >= p or s >= n):
+            return False
+
+        e = int_from_bytes(hash_sha256(sig[0:32] + bytes_from_point(P) + msg)) % n
+        c = (pow(r, 3) + 7) % p
+        y = pow(c, (p + 1) // 4, p)
+        if pow(y, 2, p) != c:
+            return False
+        R = (r, y)
+
+        a = 1 + secrets.randbelow(n-2) if i != 0 else 1
```

Comment:
This is correct but could you instead implement some method based on a deterministic RNG as described in the BIP? I think we want to encourage deterministic randomness generation because it's hard to screw up the randomness then...

Maybe we should actually add a sentence in the BIP that this is method is recommended.